### PR TITLE
Temporal improve load storage from disk

### PR DESCRIPTION
**Pull request by xrdavies**
Temporal improve load storage from disk.
Use one thread to preload storage files, one thread to add blocks to rbtree.

### CODE REVIEW

File: client/storage.c [hunk 1]
```diff
@@ -193,6 +193,304 @@ static int sort_callback(const void *l, const void *r)
...
+#else
+			sleep(.001);
+#endif
+			continue;
+		}
+				
+		uint64_t todo = item->size / sizeof(struct xdag_block);
+		uint64_t pos0 = pos;
+		
+//		memcpy(buffer, item->data, item->size);
```
Comment (jonano614) [inline client/storage.c]: Should this line be removed?

File: client/storage.c [hunk 2]
```diff
@@ -193,6 +193,304 @@ static int sort_callback(const void *l, const void *r)
...
+		struct xdag_block *buffer = item->data;
+		
+		for (i = k = 0; i < todo; ++i, pos += sizeof(struct xdag_block)) {
+			s.size += sizeof(struct xdag_block);
+//			if (buf[i].field[0].time >= start_time && buf[i].field[0].time < end_time) {
```
Comment (jonano614) [inline client/storage.c]: The same - is this line was temporary commented of should be removed?

File: client/storage.c [hunk 3]
```diff
@@ -193,6 +193,304 @@ static int sort_callback(const void *l, const void *r)
...
+};
+
+static void *init_storage_load_thread(void *data)
+{   
+	struct init_storage_param *param = (struct init_storage_param *)data;
+	xdag_time_t start_time = param->start_time;
+	xdag_time_t end_time = param->end_time;
+	
+	struct xdag_block readbuf[bufsize];
+	struct xdag_storage_sum s;
```
Comment (jonano614) [inline client/storage.c]: Could you please avoid using one-char variables? IMHO "storage_sum" is more clear

File: client/storage.c [hunk 4]
```diff
@@ -193,6 +193,304 @@ static int sort_callback(const void *l, const void *r)
...
+					s.sum += ((uint64_t*)(readbuf + i))[j];
+				}
+			}
+			
+			struct xdag_block *bs = (struct xdag_block*)malloc(sizeof(struct xdag_block)*todo);
+			memcpy(bs, readbuf, sizeof(struct xdag_block)*todo);
+			
+			while (simple_queue_push(bs, sizeof(struct xdag_block)*todo/*, path*/) == 0) {
+#if defined (_WIN32) || defined (_WIN64)
+				sleep(1);
```
Comment (jonano614) [inline client/storage.c]: Why sleep for one second?

File: client/storage.c [hunk 5]
```diff
@@ -193,6 +193,304 @@ static int sort_callback(const void *l, const void *r)
...
+		
+		if(todo > 0) {
+			for (int i = 0; i < todo; ++i) {
+				s.size += sizeof(struct xdag_block);
+				for (int j = 0; j < sizeof(struct xdag_block) / sizeof(uint64_t); ++j) {
+					s.sum += ((uint64_t*)(readbuf + i))[j];
+				}
+			}
+			
+			struct xdag_block *bs = (struct xdag_block*)malloc(sizeof(struct xdag_block)*todo);
```
Comment (jonano614) [inline client/storage.c]: Variable name "bs" is not clear.

**Comment by jonano614** [comment]
Thank you for you great job. Look at my comments please.

File: client/storage.c (hunk 1 above)
Comment (xrdavies) [reply client/storage.c]: Just keep it firstly, when finish test, everything is stable, I will remove useless comments.

File: client/storage.c (hunk 2 above)
Comment (xrdavies) [reply client/storage.c]: The same reason, I just remove the time range check when load blocks, suppose the blocks are correct. When finish testing, will remove this line. Thank you.

File: client/storage.c (hunk 3 above)
Comment (xrdavies) [reply client/storage.c]: You are right, I will rename it when test done.

File: client/storage.c (hunk 5 above)
Comment (xrdavies) [reply client/storage.c]: It should be 1ms. My mistake for windows.

File: client/storage.c (hunk 4 above)
Comment (xrdavies) [reply client/storage.c]: It should be 1ms. My mistake for windows.

**Comment by xrdavies** [comment]
@jonano614 Modified the variable names and the sleep issue.

**Pull request merged**